### PR TITLE
Restructure net.corda.core.crypto package

### PR DESCRIPTION
**Pull request by rnicoll**
Move subpackages of `net.corda.core.crypto` into the parent package, as subdividing seems to add more complexity than it reduces.

Move utility classes out of `net.corda.core.crypto` and into `net.corda.core.utilities`, except for `CryptoUtils` itself, as they are utilities that help with crypto/identity, rather than crypto themselves.

Part of https://r3-cev.atlassian.net/browse/CORDA-499

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/crypto/CompositeSignaturesWithKeys.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package net.corda.core.crypto
+
+import net.corda.core.serialization.CordaSerializable
+
+/**
+ * Custom class for holding signature data. This exists for later extension work to provide a standardised cross-platform
+ * serialization format.
+ */
+@CordaSerializable
+data class CompositeSignaturesWithKeys(val sigs: List<TransactionSignature>) {
```
Comment (kchalkias) [inline core/src/main/kotlin/net/corda/core/crypto/CompositeSignaturesWithKeys.kt]: Wasn't this removed in your previous PR?

File: core/src/main/kotlin/net/corda/core/crypto/CompositeSignature.kt [hunk 2]
```diff
@@ -1,7 +1,6 @@
-package net.corda.core.crypto.composite
+package net.corda.core.crypto
 
 import net.corda.core.crypto.SecureHash
```
Comment (kchalkias) [inline core/src/main/kotlin/net/corda/core/crypto/CompositeSignature.kt]: unused import

File: core/src/main/kotlin/net/corda/core/crypto/CordaSecurityProvider.kt [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 
-import net.corda.core.crypto.composite.CompositeKey
-import net.corda.core.crypto.composite.CompositeSignature
 import org.bouncycastle.asn1.ASN1ObjectIdentifier
 import java.security.AccessController
 import java.security.PrivilegedAction
 import java.security.Provider
 
-class CordaSecurityProvider : Provider(PROVIDER_NAME, 0.1, "$PROVIDER_NAME security provider wrapper") {
+class CordaSecurityProvider : Provider(PROVIDER_NAME, 0.1, "${PROVIDER_NAME} security provider wrapper") {
```
Comment (kchalkias) [inline core/src/main/kotlin/net/corda/core/crypto/CordaSecurityProvider.kt]: redundant curly braces?

File: core/src/main/kotlin/net/corda/core/crypto/CompositeSignaturesWithKeys.kt (hunk 1 above)
Comment (rnicoll) [reply core/src/main/kotlin/net/corda/core/crypto/CompositeSignaturesWithKeys.kt]: Sorry, that's snuck back in during the rebase

**Comment by kchalkias** [comment]
Overall it seems fine, just remove CompositeSignaturesWithKeys and will approve

**Comment by rnicoll** [comment]
I think that's everything addressed

**Pull request merged**